### PR TITLE
Issue #110 ForgeRock AM/OpenAM Security Advisory #201901-08: Open Redirect and Potential XSS

### PR DESCRIPTION
**Pull request by ogis-miyamura**
## Solution
Add redirect URL validation of OAuth 2.0 endpoint.

### CODE REVIEW

**Comment by tsujiguchitky** [comment]
> Add request parameter validation to Masthead.jsp.

This solution is for another issue(201901-03), right?

Since the commit message is rewritten when merging, please update the description of the pull request.

File: openam-oauth2/src/main/java/org/forgerock/oauth2/core/AuthorizeRequestValidatorImpl.java [hunk 1]
```diff
@@ -13,11 +13,16 @@
  *
  * Copyright 2014-2016 ForgeRock AS.
  * Portions Copyrighted 2018 Open Source Solution Technology Corporation
+ * Portions Copyrighted 2018 OGIS-RI Co., Ltd.
```
Comment (tsujiguchitky) [inline openam-oauth2/src/main/java/org/forgerock/oauth2/core/AuthorizeRequestValidatorImpl.java]: I'm sorry I didn't notice immediately.
`2018` -> `2019` ?

**Comment by ogis-miyamura** [comment]
> This solution is for another issue(201901-03), right?
> Since the commit message is rewritten when merging, please update the description of the pull request.

Sorry, it was wrong content.
Thank you for pointing out that the description should be updated.

File: openam-oauth2/src/main/java/org/forgerock/oauth2/core/AuthorizeRequestValidatorImpl.java (hunk 1 above)
Comment (ogis-miyamura) [reply openam-oauth2/src/main/java/org/forgerock/oauth2/core/AuthorizeRequestValidatorImpl.java]: Thank you for pointing out. 2019 is correct.

Comment (tsujiguchitky) [reply openam-oauth2/src/main/java/org/forgerock/oauth2/core/AuthorizeRequestValidatorImpl.java]: I'm sorry to bother you, but please check other files...

Comment (ogis-miyamura) [reply openam-oauth2/src/main/java/org/forgerock/oauth2/core/AuthorizeRequestValidatorImpl.java]: Fixed all typos.

Comment (tsujiguchitky) [reply openam-oauth2/src/main/java/org/forgerock/oauth2/core/AuthorizeRequestValidatorImpl.java]: OK, thanks!

**Comment by tsujiguchitky** [comment]
#110

**Pull request merged**